### ISSUE TITLE
PaymentTransaction.transactionIdentifier is different from ReceiptItem.transactionId

### ISSUE THREAD

**Issue by NandhakumarHCL** [bizz84/SwiftyStoreKit#302]
### Platform

- [#] iOS
- [ ] macOS
- [ ] tvOS

### In app purchase type

- [ ] Consumable
- [ ] Non-consumable 
- [#] Auto-Renewable Subscription
- [ ] Non-Renewing Subscription

### Environment

- [#] Sandbox
- [ ] Production

For Analytics i need to track the Auto-Renewable Subscription. 
Method: purchaseProduct(_ productId: String, quantity: Int = 1, atomically: Bool = true, applicationUsername: String = "", completion: @escaping (PurchaseResult) -> Void) 
- purchaseProduct returning the PurchaseResult in completion and the PurchaseResult.PurchaseDetails.transaction.transactionIdentifier is different from the ReceiptItem.transactionId.

Please provide me solution to track the Auto-Renewable Subscription in completeTransactions(atomically: Bool = true, completion: @escaping ([Purchase]) -> Void)

**Comment by bizz84** [comment]
@NandhakumarHCL Apologies for the late reply.

The problem lies in the difference between `transaction` and `originalTransaction`.

For auto-renewable subscriptions, `ReceiptItem.transactionId` refers to the payment's `originalTransaction`.

The `originalTransaction` has been available in `completeTransactions()` and `restorePurchases()` since version 0.9.0. Example:

```swift
SwiftyStoreKit.completeTransactions(atomically: true) { purchases in

    for purchase in purchases {
        switch purchase.transaction.transactionState {
        case .purchased, .restored:
            if purchase.needsFinishTransaction {
                // Deliver content from server, then:
                SwiftyStoreKit.finishTransaction(purchase.transaction)
            }
            // this will match ReceiptItem.transactionId
            print("\(purchase.originalTransaction?.transactionIdentifier)")
        case .failed, .purchasing, .deferred:
            break // do nothing
        }
    }
}
```

`originalTransaction` is not available in `purchaseProduct()`. For consistency, it would make sense to add it, right?

**Comment by bizz84** [comment]
Available on version [0.11.3](https://github.com/bizz84/SwiftyStoreKit/releases/tag/0.11.3).

Closing this issue for now, feel free to comment or reopen if needed.
